cpp/benchmarks/copying/contiguous_split_benchmark.cu

```diff
@@ -41,11 +41,8 @@ void BM_contiguous_split_common(benchmark::State& state,
   }
 
   std::vector<std::unique_ptr<cudf::column>> columns(src_cols.size());
-  std::transform(src_cols.begin(), src_cols.end(), columns.begin(), [](T& in) {
-    auto ret = in.release();
-    ret->null_count();
```

Comment:
I just happened to notice this, but that `null_count()` call has side-effects and I'm guessing that this was being done on purpose. CC @nvdbaranec

Reply:
That is what I thought as well but the functions are all marked as `const` (which is what the clang-tidy check indexes on) and the tests all came back:
```
100% tests passed, 0 tests failed out of 57
```
So if there is a side-effect, it isn't being tested.

Reply:
The `null_count` method is marked `const` but it does have a side-effect of calling a kernel. It's because the null count is treated as a cached property and is sometimes computed on otherwise `const` columns. I suspect this is done in the benchmark because computing the null count is not a part of the benchmark's target code path, and we want the property to be pre-computed so that we measure the performance of only the intended code path. Aside from pre-caching like this, there should be no reason to compute the null count until it is (lazily) requested. If this side-effect were ever needed for _correctness_ or something like thread-safety, we should reconsider our decision to implement the `null_count` method as `const`-with-side-effects, but I suspect it would cause wide-ranging breakage of `const`ness in libcudf to change that.

Reply:
The null count is internally marked as `mutable` so that it can be changed even if the column (or view) is `const`:
https://github.com/rapidsai/cudf/blob/31f92d70bdf5b80434d24b6883e45496dbe85042/cpp/include/cudf/column/column.hpp#L316
https://github.com/rapidsai/cudf/blob/31f92d70bdf5b80434d24b6883e45496dbe85042/cpp/include/cudf/column/column_view.hpp#L219

Reply:
Ok, thanks for the context. I will add back the called to `null_count` and also add some comments explaining why. :pray: @bdice